### PR TITLE
Added implementation docs for deploying to S3

### PR DESCRIPTION
**Pull request by hx-markterry**
# What does this change?

Add more info on how to implement deployment of files to S3 via CI.
# Why?

Spent a while going through this with a dev, it would be useful to others too as it is something we do a lot here.
- [x] Distribution
- [x] Shortbreaks

### CODE REVIEW

**Comment by elliottcrush** [comment]
Thanks for adding this Mark, much clearer. :+1:

File: deployment-guidelines.md [hunk 1]
```diff
@@ -38,3 +38,70 @@ Here is an example of this structure:
       hx-wizard
 
 Please raise an `INF` JIRA if you require a new S3 directory to be created, once actioned you will be given the correct credentials for CI upload your assets.
+
+## Implementation
+
+We are now using different AWS credentials for our different environments for increased security, however this mean that configuration is slightly more complex.
```
Comment (pmcnr-hx) [inline deployment-guidelines.md]: `this mean` → `this means`

File: deployment-guidelines.md [hunk 2]
```diff
@@ -38,3 +38,70 @@ Here is an example of this structure:
...
+
+First add your AWS IDs and keys as encrypted environment variables via the travis or Circle UI (not in the YAML files), the following names can be used here:
```
Comment (pmcnr-hx) [inline deployment-guidelines.md]: `travis` → `Travis` (?)

**Comment by pmcnr-hx** [comment]
Excellent resource! :+1:

File: deployment-guidelines.md [hunk 3]
```diff
@@ -38,3 +38,70 @@ Here is an example of this structure:
...
+* STAGING_AWS_ACCESS_KEY_ID
+* STAGING_AWS_SECRET_ACCESS_KEY
+
+Next create a script that uses the [AWS CLI](https://aws.amazon.com/cli/), as this is installed by default on Circle boxes and easily on Travis.
+
+The method we want to use as defined in [Continuous deployment flow](cd-flow.md), is to deploy to a staging environment when pushing to the `staging` branch and deploy to a production environment when pushing to the `master` branch.
+
+The deploy script `scripts/deploy.sh` which using the environment variable `DEPLOY_ENV` will choose the correct S3 bucket to use as this also changes.
+
+The credentials for the script will be taken from the environment variables mentioned above.
```
Comment (hx-markterry) [inline deployment-guidelines.md]: make a note about updating the credentials file on CI

File: deployment-guidelines.md [hunk 4]
```diff
@@ -38,3 +38,73 @@ Here is an example of this structure:
       hx-wizard
 
 Please raise an `INF` JIRA if you require a new S3 directory to be created, once actioned you will be given the correct credentials for CI upload your assets.
+
+## Implementation
+
+We are now using different AWS credentials for our different environments for increased security, however this mean that configuration is slightly more complex.
+
+First add your AWS IDs and keys as encrypted environment variables via the Travis or Circle UI (not in the YAML files), the following names can be used here:
```
Comment (notABluesSinger) [inline deployment-guidelines.md]: following names **should** be used

File: deployment-guidelines.md [hunk 5]
```diff
@@ -38,3 +38,73 @@ Here is an example of this structure:
...
+Next create a script that uses the [AWS CLI](https://aws.amazon.com/cli/), as this is installed by default on Circle boxes and easily on Travis.
+
+The method we want to use as defined in [Continuous deployment flow](cd-flow.md), is to deploy to a staging environment when pushing to the `staging` branch and deploy to a production environment when pushing to the `master` branch.
+
+The deploy script `scripts/deploy.sh` which using the environment variable `DEPLOY_ENV` will choose the correct S3 bucket to use as this also changes.
+
+The credentials for the script will be taken from the environment variables mentioned above and can be set via the AWS CLI as follows:
+
+    aws configure set aws_access_key_id $PRODUCTION_AWS_ACCESS_KEY_ID
+    aws configure set aws_secret_access_key $PRODUCTION_AWS_SECRET_ACCESS_KEY
```
Comment (notABluesSinger) [inline deployment-guidelines.md]: Can you put this in a code block

File: deployment-guidelines.md [hunk 6]
```diff
@@ -38,3 +38,73 @@ Here is an example of this structure:
...
+      provider: script
+      script: scripts/deploy.sh
+      on:
+        branch: staging
+        env:
+          - DEPLOY_ENV=staging
+      on:
+        branch: master
+        env:
+          - DEPLOY_ENV=production
```
Comment (notABluesSinger) [inline deployment-guidelines.md]: Code block here

File: deployment-guidelines.md [hunk 7]
```diff
@@ -38,3 +38,73 @@ Here is an example of this structure:
...
+          commands:
+            - ./scripts/deploy.sh
+            environment
+              DEPLOY_ENV: staging
+      staging:
+        branch: master
+          commands:
+            - ./scripts/deploy.sh
+            environment
+              DEPLOY_ENV: production
```
Comment (notABluesSinger) [inline deployment-guidelines.md]: Code block here

File: deployment-guidelines.md [hunk 8]
```diff
@@ -38,3 +38,73 @@ Here is an example of this structure:
...
+
+#### Local Testing
+
+For testing credentials:
+
+    aws configure
+    aws s3 cp some_test_file s3://your_bucket/your_repo_dir/ --region=eu-west-1
```
Comment (notABluesSinger) [inline deployment-guidelines.md]: Code block here

File: deployment-guidelines.md [hunk 9]
```diff
@@ -38,3 +38,73 @@ Here is an example of this structure:
...
+
+The deploy script can be run as follows:
+
+    STAGING_AWS_ACCESS_KEY_ID=your_id STAGING_AWS_SECRET_ACCESS_KEY=your_key DEPLOY_ENV=staging ./scripts/deploy.sh
```
Comment (notABluesSinger) [inline deployment-guidelines.md]: Code block here

**Pull request merged**